core/src/main/java/io/questdb/griffin/engine/functions/cast/CastCharToTimestampFunctionFactory.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.cast;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.AbstractUnaryTimestampFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+
+public class CastCharToTimestampFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "cast(An)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) {
+        return new CastCharToTimestampFunctionFactory.Func(args.getQuick(0));
+    }
+
+    private static class Func extends AbstractUnaryTimestampFunction {
+        public Func(Function arg) {
+            super(arg);
+        }
+
+        @Override
+        public long getTimestamp(Record rec) {
+            final byte v = (byte) (arg.getChar(rec) - '0');
```

Comment:
should use `implicitCastCharAsType` here to reduce logic duplication

Reply:
I don't know. Implicit casting throws an exception when casting is impossible while explicit casts are more lenient. This is the case for other already existing casts. 

So to re-use implicit casting methods we would need to catch the `ImplicitCastingException` and return `Numbers.LONG_NaN`. 

`ImplicitCastingException` is not cheap - it's crafting a message string. Doing it only to catch it and swallow seems like a waste.